include/aws/sdkutils/private/endpoints_types_impl.h

```diff
@@ -115,13 +118,13 @@ struct aws_endpoints_rule_data_tree {
 
 struct aws_endpoints_condition {
     struct aws_endpoints_function function;
-    struct aws_string *assign;
+    struct aws_byte_cursor assign;
 };
 
 struct aws_endpoints_rule {
     /* List of (aws_endpoints_condition) */
     struct aws_array_list conditions;
-    struct aws_string *documentation;
+    struct aws_byte_cursor documentation;
```

Comment:
similar question: these cursors are pointing to safe memory that always outlives them?

Reply:
yes